File: Stacks/stack.c
#include<stdio.h>
#include<stdlib.h>
#include<stdbool.h>

struct Node{
    int data;
    struct Node * next;
};

struct Node * head =NULL;

void push(int data){
    
    struct Node * current=head;
    struct Node * newNode=(struct Node *)malloc(sizeof(struct Node));
    newNode->data=data;
    newNode->next=current;
    head=newNode;

}
bool isStackEmpty(){
    struct Node * current=head;
    if(current==NULL){   
        return true;
    }
    return false;
}
int pop(){
    struct Node * current=head;
    struct Node * temp=NULL;
    if(isStackEmpty()){
        printf("%s","stack is empty");
        return 0;
    }
    temp=current;
    current=current->next;
    head=current;
    free(temp);
}

int Top(){
    return head->data;
}

void stackElements(){
    struct Node * current=head;
    while(current!=NULL){
        
        printf("%d",current->data);
        current=current->next;
    }
}

int main(){
    push(1);
    push(2);
    push(3);
    pop();
    stackElements();
}

File: Stacks/stack.py
class Node:
    def __init__(self,data):
        self.data=data
        self.next=None
class Stack:
    def __init__(self):
        self.top=None
    def push(self,data):
        newNode=Node(data)
        newNode.next=self.top
        self.top=newNode
    def isEmpty(self):
        return self.top==None
    def pop(self):
        current=self.top
        temp=current
        current=current.next
        self.top=current
        del(temp)
    def stackElements(self):
        current=self.top
        if(self.isEmpty()):
            print("Stack is empty")
            return
        while(current):
            print(current.data," ")
            current=current.next

if __name__=="__main__":
    s=Stack()
    s.push(5)
    s.push(4)
    s.push(3)
    s.push(2)
    s.push(1)
    
    s.stackElements()
    s.pop()
    print("after pop:")
    s.stackElements()